### PR TITLE
Loading license data from files in Java and C#

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Also:

* Having SurfaceNamer store packageName
* Restructuring top-level views for Java and C# so that they are better
  normalized (file-level information is separate from class information)

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java [hunk 1]
```diff
@@ -68,7 +72,7 @@
   private ApiMethodTransformer apiMethodTransformer;
   private PageStreamingTransformer pageStreamingTransformer;
   private BundlingTransformer bundlingTransformer;
-  private ImportTypeTransformer importTypeTransformer;
+  private FileHeaderTransformer fileHeaderTransformer;
   private RetryDefinitionsTransformer retryDefinitionsTransformer;
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java]: A general question: Should these be turned into static finals to help signify that they don't hold state?

File: src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java [hunk 2]
```diff
@@ -212,17 +214,22 @@ private PackageInfoView generatePackageInfo(
...
-        .importPath(apiConfig.getPackageName())
-        .packageName(namer.getLocalPackageName())
-        .serviceDocs(Collections.<ServiceDocView>emptyList())
-        // TODO(pongad): GoContextCommon should be deleted after we convert Go discovery to MVMM.
-        .packageDoc(new GoContextCommon().getWrappedCommentLines(doc))
-        .domainLayerLocation(apiConfig.getDomainLayerLocation())
-        .build();
+    PackageInfoView.Builder packageInfo = PackageInfoView.newBuilder();
+
+    packageInfo.templateFileName(DOC_TEMPLATE_FILENAME);
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java]: Nothing wrong with this, but wondering why you're against using the chained calls?

File: src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java]: Even static final objects can have state. Making them at least `final` makes sense though. Maybe in a separate PR?

File: src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java [hunk 3]
```diff
@@ -155,23 +170,15 @@ private StaticLangXApiView generateXApi(SurfaceTransformerContext context) {
...
   private boolean methodTypeHasImpl(ApiMethodType type) {
     return type != ApiMethodType.FlattenedAsyncCancellationTokenMethod;
   }
 
-  private StaticLangXSettingsView generateXSettings(SurfaceTransformerContext context) {
-    StaticLangXSettingsView.Builder xsettingsClass = StaticLangXSettingsView.newBuilder();
-    xsettingsClass.templateFileName(""); // Unused in C#
-    xsettingsClass.packageName(context.getApiConfig().getPackageName());
+  private StaticLangSettingsView generateSettingsClass(SurfaceTransformerContext context) {
+    StaticLangSettingsView.Builder xsettingsClass = StaticLangSettingsView.newBuilder();
```
Comment (chrisdunelm) [inline src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java]: Remove x from variable name?

**Comment by michaelbausor** [comment]
LGTM

**Pull request merged**